lib/handler/RetryHandler.js

```diff
@@ -0,0 +1,347 @@
+const assert = require('node:assert')
+
+const { kRetryHandlerDefaultRetry } = require('../core/symbols')
+const { RequestRetryError } = require('../core/errors')
+const {
+  isDisturbed,
+  parseHeaders,
+  parseRangeHeader,
+  safeHTTPMethods
+} = require('../core/util')
+
+function calculateRetryAfterHeader (retryAfter) {
+  const current = Date.now()
+  const diff = new Date(retryAfter).getTime() - current
+
+  return diff
+}
+
+class RetryHandler {
+  constructor (opts, handlers, retryOpts) {
```

Comment:
I would expect things to live inside opts, i.e. no retryOpts.

```suggestion
  constructor ({ retry: retryOpts, ...opts }, handlers) {
    if (typeof retryOpts === 'function') {
      retryOpts = { retry: retryOpts }
    } else if (typeof retryOpts === 'number') {
      retryOpts = { maxRetries: retryOpts }
    }
  }
```

Reply:
Of course this has the downside of potential conflict with other handlers... so maybe a separate handler opts is not bad either.

Reply:
TBH, had the same thoughts as you, but the conflict won on my side so I decided to go with the weird opts. I'd prefer it to keep it there but maybe change the order first `opts` and later `handlers`?

Just wanted to avoid collisions, specially as the `opts` are reused between retries for the `dispatch`

Reply:
Would you be fine with my suggestion?

Reply:
Sure thing 👍